src/jltypes.c

```diff
@@ -3747,24 +3748,30 @@ void jl_init_types(void)
 
     jl_typemap_entry_type =
         jl_new_datatype(jl_symbol("TypeMapEntry"), jl_any_type, jl_emptysvec,
-                        jl_svec(9, jl_symbol("next"),
-                                   jl_symbol("sig"),
-                                   jl_symbol("tvars"),
-                                   jl_symbol("simplesig"),
-                                   jl_symbol("guardsigs"),
-                                   jl_symbol("func"),
-                                   jl_symbol("isleafsig"),
-                                   jl_symbol("issimplesig"),
-                                   jl_symbol("va")),
-                        jl_svec(9, jl_any_type, // Union{TypeMapEntry, Void}
-                                   jl_type_type, // TupleType
-                                   jl_any_type, // Union{SimpleVector{TypeVar}, TypeVar}
-                                   jl_any_type, // TupleType
-                                   jl_any_type, // SimpleVector{TupleType}
-                                   jl_any_type, // Any
-                                   jl_bool_type,
-                                   jl_bool_type,
-                                   jl_bool_type),
+                        jl_svec(11,
+                            jl_symbol("next"),
+                            jl_symbol("sig"),
+                            jl_symbol("tvars"),
+                            jl_symbol("simplesig"),
+                            jl_symbol("guardsigs"),
+                            jl_symbol("min-world"),
+                            jl_symbol("max-world"),
```

Comment:
Should be underscores. True, we don't want anybody messing with these, but I think experience has shown that "hidden" fields have been more trouble than they're worth.